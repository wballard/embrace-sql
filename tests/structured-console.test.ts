import stdMocks from "std-mocks";
import sc from "../src/structured-console";

describe("structured console logging", () => {
  it("exists", () => {
    expect(sc).toBeTruthy();
  });
  it("captures log", async () => {
    stdMocks.use();
    console.log("Hello World");
    stdMocks.restore();
    expect(stdMocks.flush()).toMatchSnapshot();
  });
  it("captures debug", async () => {
    stdMocks.use();
    console.debug("Hello World");
    stdMocks.restore();
    expect(stdMocks.flush()).toMatchSnapshot();
  });
  it("captures info", async () => {
    stdMocks.use();
    console.info("Hello World");
    stdMocks.restore();
    expect(stdMocks.flush()).toMatchSnapshot();
  });
  it("captures warn", async () => {
    stdMocks.use();
    console.warn("Hello World");
    stdMocks.restore();
    expect(stdMocks.flush()).toMatchSnapshot();
  });
  it("captures error", () => {
    stdMocks.use();
    console.error("Hello World");
    stdMocks.restore();
    expect(stdMocks.flush()).toMatchSnapshot();
  });
});
