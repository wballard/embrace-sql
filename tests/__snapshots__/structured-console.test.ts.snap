// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`structured console logging captures debug 1`] = `
Object {
  "stderr": Array [],
  "stdout": Array [
    "{\\"logLevel\\":\\"debug\\",\\"source\\":\\"/Users/wballard/embrace-sql-core/tests/structured-console.test.ts#21\\",\\"message\\":\\"Hello World\\",\\"additional\\":[[]]}",
  ],
}
`;

exports[`structured console logging captures error 1`] = `
Object {
  "stderr": Array [
    "{\\"logLevel\\":\\"error\\",\\"source\\":\\"/Users/wballard/embrace-sql-core/tests/structured-console.test.ts#39\\",\\"message\\":\\"Hello World\\",\\"additional\\":[[]]}",
  ],
  "stdout": Array [],
}
`;

exports[`structured console logging captures info 1`] = `
Object {
  "stderr": Array [],
  "stdout": Array [
    "{\\"logLevel\\":\\"info\\",\\"source\\":\\"/Users/wballard/embrace-sql-core/tests/structured-console.test.ts#27\\",\\"message\\":\\"Hello World\\",\\"additional\\":[[]]}",
  ],
}
`;

exports[`structured console logging captures log 1`] = `
Object {
  "stderr": Array [],
  "stdout": Array [
    "{\\"logLevel\\":\\"info\\",\\"source\\":\\"/Users/wballard/embrace-sql-core/tests/structured-console.test.ts#15\\",\\"message\\":\\"Hello World\\",\\"additional\\":[[]]}",
  ],
}
`;

exports[`structured console logging captures warn 1`] = `
Object {
  "stderr": Array [],
  "stdout": Array [
    "{\\"logLevel\\":\\"warn\\",\\"source\\":\\"/Users/wballard/embrace-sql-core/tests/structured-console.test.ts#33\\",\\"message\\":\\"Hello World\\",\\"additional\\":[[]]}",
  ],
}
`;
